### PR TITLE
feat: enable otp email and sms

### PR DESCRIPTION
**Pull request by eliobischof**
- Adds the possibility to add and remove Email OTP and Password OTP as second factors via API.
- Internally renames the existing OTP type to TOTP (the APIs are backwards compatible)
- Adds outcommented lines in instance setup to add the factors to the login policy (https://github.com/zitadel/zitadel/pull/6260#discussion_r1273549314)
- removes some unused code (https://github.com/zitadel/zitadel/pull/6260#discussion_r1273522093, https://github.com/zitadel/zitadel/pull/6260#discussion_r1273522244)
- removes some useless tests (https://github.com/zitadel/zitadel/pull/6260#discussion_r1273520753)

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/api/grpc/settings/v2/settings_converter_test.go [hunk 1]
```diff
@@ -83,122 +85,6 @@ func Test_loginSettingsToPb(t *testing.T) {
...
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.want.String(), func(t *testing.T) {
-			got := multiFactorTypeToPb(tt.args.typ)
-			assert.Equal(t, tt.want, got)
-		})
-	}
-}
-
```
Comment (eliobischof) [inline internal/api/grpc/settings/v2/settings_converter_test.go:201]: These tests have no value, as they use the same algorithm like the code, which is also just a hardcoded mapping.

File: internal/domain/factors.go [hunk 2]
```diff
@@ -4,20 +4,14 @@ type SecondFactorType int32
...
 )
 
-func SecondFactorTypes() []SecondFactorType {
-	types := make([]SecondFactorType, 0, secondFactorCount-1)
-	for i := SecondFactorTypeUnspecified + 1; i < secondFactorCount; i++ {
-		types = append(types, i)
-	}
-	return types
-}
-
```
Comment (eliobischof) [inline internal/domain/factors.go:20]: unused

File: internal/command/instance.go [hunk 3]
```diff
@@ -241,8 +241,12 @@ func (c *Commands) SetUpInstance(ctx context.Context, setup *InstanceSetup) (str
 			setup.LoginPolicy.SecondFactorCheckLifetime,
 			setup.LoginPolicy.MultiFactorCheckLifetime,
 		),
-		prepareAddSecondFactorToDefaultLoginPolicy(instanceAgg, domain.SecondFactorTypeOTP),
+		prepareAddSecondFactorToDefaultLoginPolicy(instanceAgg, domain.SecondFactorTypeTOTP),
 		prepareAddSecondFactorToDefaultLoginPolicy(instanceAgg, domain.SecondFactorTypeU2F),
+		/* TODO: incomment when usable
+		prepareAddSecondFactorToDefaultLoginPolicy(instanceAgg, domain.SecondFactorTypeOTPEmail),
+		prepareAddSecondFactorToDefaultLoginPolicy(instanceAgg, domain.SecondFactorTypeOTPSMS),
+		*/
```
Comment (eliobischof) [inline internal/command/instance.go:249]: This should be incommented here https://github.com/zitadel/zitadel/issues/6207#issuecomment-1649848758

File: internal/api/grpc/settings/v2/settings_converter_test.go (hunk 1 above)
Comment (livio-a) [reply internal/api/grpc/settings/v2/settings_converter_test.go:201]: i think this is just for ensuring that all types are correctly returned and unknown ignored

Comment (livio-a) [reply internal/api/grpc/settings/v2/settings_converter_test.go:201]: and we could handle the transition from OTP to TOTP here as well

**Review by livio-a — COMMENTED** [review]
@eliobischof the new types were not available in console... i've added them
please check if this is ok

File: internal/command/instance.go (hunk 3 above)
Comment (livio-a) [reply internal/command/instance.go:249]: ok for me... but is it necessary as you can enable them anyway? (and still not be able to use them)

Comment (eliobischof) [reply internal/command/instance.go:249]: Wouldn't that be unnecessarily confusing?

File: proto/zitadel/settings/v2alpha/login_settings.proto [hunk 4]
```diff
@@ -113,8 +113,11 @@ message LoginSettings {
 
 enum SecondFactorType {
   SECOND_FACTOR_TYPE_UNSPECIFIED = 0;
+  // This is the type for TOTP
   SECOND_FACTOR_TYPE_OTP = 1;
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply proto/zitadel/settings/v2alpha/login_settings.proto:117]: I tried to rename it but the buf action didn't let me break it :-)

[the remark this answers is not in the document]
Comment (eliobischof) [reply proto/zitadel/settings/v2alpha/login_settings.proto:117]: I can try again

File: internal/api/grpc/settings/v2/settings_converter_test.go (hunk 1 above)
Comment (eliobischof) [reply internal/api/grpc/settings/v2/settings_converter_test.go:201]: It is exactly the same algorithm, so it is a pseudo test. This is also true for the TOTP and the unspecified mapping.
I'd be happy with a test that ensures that no type mapping is forgotten and unknown types map to unspecified (I can do that).
Do you see my point?

**Comment by eliobischof** [comment]
> @eliobischof the new types were not available in console... i've added them please check if this is ok

What do you mean and where did you add something?

**Comment by livio-a** [comment]
> > @eliobischof the new types were not available in console... i've added them please check if this is ok
> 
> What do you mean and where did you add something?

sorry messed up the push... it is now 😃

File: internal/api/grpc/settings/v2/settings_converter_test.go (hunk 1 above)
Comment (livio-a) [reply internal/api/grpc/settings/v2/settings_converter_test.go:201]: what do you mean by same algorithm? it's a function with an input and an output, so classic unit test.
i'm aware that it feels kind of senseless mapping `domain.A` to `proto.A` but still there might be functions where `proto.A` and `proto.1` must be mapped to `domain.B` or vice versa.
BTW the linter will make sure that every type is used in the switch.

Comment (eliobischof) [reply internal/api/grpc/settings/v2/settings_converter_test.go:201]: By same algorithm I mean youe just map each type to another type manually 1:1. The algorithm is in the coders brain in both the program and the test.

I'll add the tests again, we can discuss it another time 😄

**Review by livio-a — APPROVED** [review]

**Review by S-code652 — APPROVED** [review]

**Pull request merged**